### PR TITLE
Use real-time database in user module

### PR DESCRIPTION
**Pull request by cp-pratik-k**
### Purpose
 - Use real-time database in user module

### Summary of Changes
 - Use realtime database in user home, leave and members screen.

### Test steps
 - run flutter test

### Conformity
- [x] Followed [git guidelines](https://github.com/canopas/flutter-developer-roadmap/blob/main/GitGuideline.md) for creating commit messages and Pull Request guidelines.
- [x] Self-approved the PR - reviewed the PR as a reviewer and gave it self-approval if everything is ok. If not, made the required changes.
- [x] Ensured that the PR satisfies all specified requirements in the ticket, including bug fixes and new features.
- [x] Provided test steps, including steps to reproduce the issue or test the new functionality, ensuring other team members can verify the changes.
- [x] Added/Updated proper code comments to make it easy-to-understand for other developers.
- [x] Reused code (if the same code was written twice, made it common and reused it at both places).
- [x] Removed unused or commented code if not required.
- [x] Ensured proper Dart naming conventions were used for variables, classes, and methods.
- [x] Localized user-facing strings.
- [x] Included screenshots/videos of behavior changes: Provided visual evidence of any changes to UI or behavior for easier review and understanding in the PR description.
- [x] Implemented proper error handling: Ensured that the code anticipated and handled potential errors and edge cases gracefully.
- [x] Avoided introducing technical debt: If the PR introduces technical debt, created and linked appropriate tickets for future resolution.
- [x] Included relevant unit tests: Wrote unit tests that focused on testing behavior and functionality, rather than merely covering lines of code.
- [x] Ensured code was performant and scalable: Verified that the changes did not introduce performance issues or bottlenecks and could scale as needed.
- [x] Ensured comments were up-to-date and relevant to the code to describe complex logic and to add understanding for other developers.
- [x] Marked the PR as ready before submitting it for review.

### CODE REVIEW

File: lib/ui/user/home/home_screen/bloc/user_home_bloc.dart [hunk 1]
```diff
@@ -1,43 +1,49 @@
...
+    on<ShowError>(_showError);
+    on<UpdateLeaveRequest>(_updateLeaveRequest);
+    add(ShowLoading());
+    _leaveRequestStreamSubscription = _leaveService
+        .getRequestedLeaveSnapshot(_userManager.employeeId)
+        .listen((request) {
+      add(UpdateLeaveRequest(request));
+    }, onError: (error, _) {
+      add(ShowError(firestoreFetchDataError));
     });
```
Comment (cp-sneha-s) [inline lib/ui/user/home/home_screen/bloc/user_home_bloc.dart:28]: Let's add log in to the firebase crashlytics in case of error

Comment (cp-pratik-k) [reply lib/ui/user/home/home_screen/bloc/user_home_bloc.dart:28]: what if its internet error?

**Comment by cp-sneha-s** [comment]
- Repository layer is not added.
- Subscribe to the same stream multiple times.
- Will work on the same functionality by implementing Stream.
